src/ast/semantic_analyser.cpp

```diff
@@ -1725,6 +1793,34 @@ void SemanticAnalyser::visit(AttachPoint &ap)
       }
     }
   }
+  else if (ap.provider == "kfunc" || ap.provider == "kretfunc")
+  {
+#ifdef HAVE_KFUNC
+    bool supported = bpftrace_.btf_.has_data();
+#else
+    bool supported = false;
+#endif
+
+    if (!supported)
+    {
+      error("kfunc/kretfunc not available for your kernel version.", ap.loc);
```

Comment:
As a user, do you think it'd be helpful to see an error message about BTF? Or is that too implementation specific?

Reply:
user just used 'kfunc or kretfunc' so the message seems related,
maybe it'd more confusing mentioning BTF, not sure ;-)